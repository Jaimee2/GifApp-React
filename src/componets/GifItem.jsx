import React from 'react';

function GifItem({img}) {
    return (
        <div className="card">
            <img src={ img.url } alt="Alt title "/>
            <p>{ img.title }</p>
        </div>
    );
}

export default GifItem;