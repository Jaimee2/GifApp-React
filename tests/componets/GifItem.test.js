import { render, screen } from "@testing-library/react";
import GifItem from "../../src/componets/GifItem.jsx";

describe('GiftItemTest', () => {

    const url = 'http://...';
    const title = 'Title of the gif';

    test('Snapshot test', () => {
        const {container} = render(<GifItem img={ {url, title} }/>)
        expect(container).toMatchSnapshot();
    });

    test('Snapshot test', () => {

        render(<GifItem img={ {url, title} }/>)

        const {src, alt} = screen.getByRole('img');
        screen.debug();

        expect(src).toBe(url + "/");
        expect(alt).toBe(title);
    });


});