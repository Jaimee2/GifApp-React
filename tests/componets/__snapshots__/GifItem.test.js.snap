// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftItemTest Snapshot test 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Title of the gif"
      src="http://..."
    />
    <p>
      Title of the gif
    </p>
  </div>
</div>
`;
